maths/nth_sgonal_num.py

```diff
@@ -0,0 +1,24 @@
+def nth_sgonal_num(num: int, sides: int) -> int:
+    """
+    Returns the `num`th `sides`-gonal number. It is assumed that `num` >= 0 and
+    `sides` >= 3 (see for reference https://en.wikipedia.org/wiki/Polygonal_number).
+
+    >>> nth_sgonal_num(0, 3)
+    0
+    >>> nth_sgonal_num(3, 3)
+    6
+    >>> nth_sgonal_num(5, 4)
+    25
+    >>> nth_sgonal_num(2, 5)
+    5
+    """
```

Comment:
assuming num>=0 & sides>=3 is ok but what if someone entered some other cases, your code should be able to handle that case, so add more doctest like `>>> nth_sgonal_num(0,0)`, `>>> nth_sgonal_num(0,a)`, `>>> nth_sgonal_num(0,1)`, `>>> nth_sgonal_num(1,1)`